packages/blade/src/tokens/theme/createTheme.ts

```diff
@@ -0,0 +1,379 @@
+  for (let i = 0; i < 6; i++) {
+    currentColor = currentColor.brighten(lightnessFactor);
+    palette.push(currentColor.toHslString());
+  }
+
+  currentColor = tinycolor(baseColorHslString); // Reset to the base color
+
+  // Generate shades darker
+  for (let i = 0; i < 4; i++) {
+    currentColor = currentColor.darken(darknessFactor);
+    palette.unshift(currentColor.toHslString()); // Add shades at the beginning of the palette
+  }
+
+  const colorPalette = palette.reverse();
+  const brandPrimaryColor = colorPalette[6];
+
+  const brandColors = {
+    '50': colorPalette[0],
+    '100': colorPalette[1],
+    '200': colorPalette[2],
+    '300': colorPalette[3],
+    '400': colorPalette[4],
+    '500': colorPalette[5],
+    '600': brandPrimaryColor,
+    '700': colorPalette[7],
+    '800': colorPalette[8],
+    '900': colorPalette[9],
+    '950': colorPalette[10],
+    a00: getColorWithOpacity(brandPrimaryColor, opacity[0]),
+    a50: getColorWithOpacity(brandPrimaryColor, opacity[1]),
+    a100: getColorWithOpacity(brandPrimaryColor, opacity[2]),
+    a200: getColorWithOpacity(brandPrimaryColor, opacity[3]),
+  };
+
+  return brandColors;
+};
+
+/**
+ *
+ * @param brandColors - The brand colors to use to override the payment theme
+ * @description Overrides the payment theme with the brand colors passed in
+ * @returns The light theme tokens with the custom brand colors
+ */
+const getLightTheme = (brandColors: Color['chromatic']['azure']): ThemeTokens => {
+  // Select the most readable color to use as the foreground color on top of brand color
+  // For example: On Primary Button where the background color is brand color, the text color should be either dark or light depending on which is more readable on top of that brand color
+  const foregroundOnBrandColorPaymentLight = tinycolor
+    .mostReadable(
+      brandColors[900],
+      [globalColors.neutral.blueGrayLight[1100], globalColors.neutral.blueGrayLight[50]],
+      WCAG2ContrastOptions,
+    )
+    .toHexString();
+
+  // Select the most readable color to use as the foreground color on top of surface color
+  // For example: On Secondary Button where the background color is surface color, the text color should be either the brand color or dark color depending on which is more readable on top of that surface color
+  const foregroundOnSurfacePaymentLight = tinycolor.isReadable(
+    globalColors.neutral.blueGrayLight[50],
+    brandColors[600],
+    WCAG2ContrastOptions,
+  )
+    ? brandColors[600]
+    : globalColors.neutral.blueGrayLight[1100];
+
+  // Override the payment theme onLight with the brand colors
+  const lightThemePaymentsOverrides = {
+    colors: {
+      onLight: {
+        brand: {
+          primary: {
+            300: brandColors.a50,
+            400: brandColors.a100,
+            500: brandColors[600],
+            600: brandColors[700],
+            700: brandColors[800],
+            800: brandColors[950],
+          },
+          gray: {
```

Comment:
what about other grays? 300, 400 etc?

Reply:
Grays we're scoping out of our solution for now since grays don't have a major impact but in a very specific case for checbox & radio, gray had to be handled to manage contrast - which is only this gray.

In future if we see grays breaking our solution, we can take up branding gray colors as well